src/track/beatutils.cpp

```diff
@@ -289,16 +289,22 @@ double BeatUtils::makeConstBpm(
 
     // Create a const region region form the first beat of the first region to the last beat of the last region.
 
-    const double minRoundBpm = 60 * sampleRate / longestRegionBeatLengthMax;
-    const double maxRoundBpm = 60 * sampleRate / longestRegionBeatLengthMin;
-    const double centerBpm = 60 * sampleRate / longestRegionBeatLength;
+    const double minRoundBpm = 60.0 * sampleRate / longestRegionBeatLengthMax;
+    const double maxRoundBpm = 60.0 * sampleRate / longestRegionBeatLengthMin;
+    const double centerBpm = 60.0 * sampleRate / longestRegionBeatLength;
 
     //qDebug() << "minRoundBpm" << minRoundBpm;
     //qDebug() << "maxRoundBpm" << maxRoundBpm;
     const double roundBpm = roundBpmWithinRange(minRoundBpm, centerBpm, maxRoundBpm);
 
     if (pFirstBeat) {
+        // Move the first beat as close to the start of the track as we can. This is
+        // a constant beatgrid so "first beat" only affects the anchor point where
+        // bpm adjustments are made.
```

Comment:
Please add a note, that this is only a temporary solution that will be replaced with the first down-beat.

We can already consider to use the first beat after the silence period or an unusable noise without a rhythm  over.
But than this is clearly no longer a last minute bug fix. So this is fine for now.

Reply:
> Please add a note, that this is only a temporary solution that will be replaced with the first down-beat.

I don't know if I understand this correctly, but keep in mind that not every song starts with a downbeat, e.g. [Prince - I Wanna Be Your Lover](https://www.youtube.com/watch?v=Rp8WL621uGM) starts with the beat before the downbeat.

Reply:
Should we stick with the first beat for now to restore the previous behaviour?
That being the anchor is currently the most predictable behaviour IMO.

We could use all sorts of anchors here, as long as we mention that somewhere (tooltip, manual)
* first beat after silence¹
* intro_end²
* Cue point³ (my preference btw)

while ¹² might be wrong/unreliable because
¹ might be on an offbeat of unconst drum intro for an otherwise const track
² intro_end might not be set
³ might work in most cases if we consider the Cue is set close to the start while anylyzing and preparing a track.

I suppose all that can be evaluated properly once we have the new beats merged.

Reply:
Should we stick with the first beat for now to restore the previous behaviour?
That being the anchor is currently the most predictable behaviour IMO.

We could use all sorts of anchors here, as long as we mention that somewhere (tooltip, manual)
* first beat after silence¹
* intro_end²
* Cue point³ (my preference btw)

while ¹² might be wrong/unreliable because
¹ might be on an offbeat of unconst drum intro for an otherwise const track
² intro_end might not be set
³ might work in most cases if we consider the Cue is set close to the start while anylyzing and preparing a track.

I suppose all that can be evaluated properly once we have the new beats merged.

Reply:
You can add the fist bar after silence here.

However this question is kind of petty here, because the point declared as the first beat is kind of random here in a failed detection.

Before this patch, we take the first beat of the longest const region, which is an actual onset.

Now we take a calculated beat extended into the region before the const region, where the onsets are likely off. However this does not matter, because the user is not satisfied with the detection anyway and needs an tempo adjustment. 

@ywwg What was the track, where you noticed the issue first? Since the new detection I had never the need to adjust the BPM by small amounts. So it might be an edge case that reveals a bug in the detection itself.

Reply:
This was happening in the majority of new tracks I downloaded.

Eventaully, the anchor point should be the downbeat nearest to the Cue point.  It doesn't have to be 100% perfect!  But to get the release out, I think the earliest downbeat is the best solution and is most consistent with previous Mixxx behavior

Reply:
Also remember that bpm adjustment is extremely hard to see near the anchor point -- whether the point is at the very first beat, or at the first downbeat, is basically impossible to tell the difference.  We can try to address the larger question in subsequent work, but that shouldn't be done in this PR -- let's take it to Zulip instead.  I think the only relevant question here is, is this fix sufficient to work around the problem in 2.3 so we can do the release.  I'll add a comment that this is a temporary fix.